src/tracing/builder/parity.rs

```diff
@@ -249,15 +249,15 @@ impl ParityTraceBuilder {
                     }
                 }
             }
-        }
 
-        // Sort the traces only if a selfdestruct trace was encountered
-        if sorting_selfdestruct {
-            traces.sort_by(|a, b| a.trace_address.cmp(&b.trace_address));
-        }
+            // Sort the traces only if a selfdestruct trace was encountered
+            if sorting_selfdestruct {
+                traces.sort_by(|a, b| a.trace_address.cmp(&b.trace_address));
```

Comment:
this can be sort_by_key

Reply:
Seems like `sort_by_key` needs to consume the Key, ref https://users.rust-lang.org/t/why-does-slice-sort-by-key-uses-b-and-not-b-as-the-key/15781

this will need to copy the `trace_address`

```
error[E0507]: cannot move out of `t.trace_address` which is behind a shared reference
   --> src/tracing/builder/parity.rs:255:40
    |
255 |                 traces.sort_by_key(|t| t.trace_address);
    |                                        ^^^^^^^^^^^^^^^ move occurs because `t.trace_address` has type `Vec<usize>`, which does not implement the `Copy` trait
    |
help: consider cloning the value if the performance cost is acceptable
    |
255 |                 traces.sort_by_key(|t| t.trace_address.clone());
    |                                                       ++++++++

```

Reply:
yeah references also don't work for sort by key